armi/reactor/zones.py

```diff
@@ -98,9 +96,32 @@ def addLoc(self, loc: str) -> None:
         assert isinstance(loc, str), "The location must be a str: {0}".format(loc)
         self.locs.add(loc)
 
+    def removeLoc(self, loc: str) -> None:
+        """
+        Removes the location from this Zone.
+
+        Parameters
+        ----------
+        loc : str
+            Location within the Core.
+
+        Notes
+        -----
+        This method does not validate that the location given is somehow "valid".
+        We are not doing any reverse lookups in the Reactor to prove that the type
+        or location is valid. Because this would require heavier computation, and
+        would add some chicken-and-the-egg problems into instantiating a new Reactor.
+
+        Returns
+        -------
+        None
+        """
+        assert isinstance(loc, str), "The location must be a str: {0}".format(loc)
```

Comment:
You should raise an Exception here. Also it's better to check that the location is the correct format as well.

Reply:
We're currently not validating the locations, but this may be enhanced in the future:
https://github.com/terrapower/armi/blob/ffda4f1850f0835f4f1d95f383bea9dd1f4fdbde/armi/reactor/zones.py#L54

I used `assert` here to maintain consistency with the other `zones` functions. Is it fine to leave this as-is, or should the other functions be changed as well?

Reply:
It's the same, but for consistency just raising an Exception is good with a message.